internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -435,6 +450,36 @@ func resourceDatabricksWorkspaceCreateUpdate(d *pluginsdk.ResourceData, meta int
 		}
 	}
 
+	diskKeyIdRaw := d.Get("managed_disk_cmk_key_vault_key_id").(string)
+	if diskKeyIdRaw != "" {
+		setEncrypt = true
+		key, err := keyVaultParse.ParseNestedItemID(diskKeyIdRaw)
+		if err != nil {
+			return err
+		}
+
+		encrypt.Entities.ManagedDisk = &workspaces.ManagedDiskEncryption{
+			// There is only one valid source for this field at this point in time so I have hardcoded the value
+			KeySource: workspaces.EncryptionKeySourceMicrosoftPointKeyvault,
+			KeyVaultProperties: workspaces.ManagedDiskEncryptionKeyVaultProperties{
+				KeyName:     key.Name,
+				KeyVersion:  key.Version,
+				KeyVaultUri: key.KeyVaultBaseUrl,
+			},
+		}
+
+		rotationEnabled := d.Get("managed_disk_cmk_rotation_to_latest_version_enabled").(bool)
+		if rotationEnabled {
+			encrypt.Entities.ManagedDisk.RotationToLatestKeyVersionEnabled = utils.Bool(rotationEnabled)
+		}
+
+		// make sure the key vault exists
+		keyVaultIdRaw, err := keyVaultsClient.KeyVaultIDFromBaseUrl(ctx, meta.(*clients.Client).Resource, key.KeyVaultBaseUrl)
+		if err != nil || keyVaultIdRaw == nil {
+			return fmt.Errorf("retrieving the Resource ID for the customer-managed keys for managed disk Key Vault at URL %q: %+v", key.KeyVaultBaseUrl, err)
+		}
+	}
```

Comment:
Should this happen further up i.e. as soon as we know that a user wants to configure this?

Reply:
Could move it up, although functionality wise it doesn't really change anything :)